app/feed/page.tsx

```diff
@@ -29,15 +49,46 @@ export default async function FeedPage({ searchParams }: Props) {
   if (!Object.entries(events).length) {
     return <LoadingText text="Fetching latest events" />;
   }
+
+  if (Repository && Events) {
+    events = events.filter((event) => {
+      const repoName = event.repo.name.replace(
+        `${env.NEXT_PUBLIC_GITHUB_ORG}/`,
+        "",
+      );
+      return (
+        repoName === searchParams.Repository &&
+        event.type === searchParams.Events
+      );
+    });
+  } else if (Repository) {
+    events = events.filter((event) => {
+      const repoName = event.repo.name.replace(
+        `${env.NEXT_PUBLIC_GITHUB_ORG}/`,
+        "",
+      );
+      return repoName === searchParams.Repository;
+    });
+  } else if (Events) {
+    events = events.filter((event) => {
+      return event.type === searchParams.Events;
+    });
+  }
```

Comment:
Code is complicated than necessary.

Concatenate the filters.

First filter by selected repository and then filter by events...

```js
let resultSet = allEvents;

resultSet = resultSet.filter(FilterBySelectedRepositories);
resultSet = resultSet.filter(FilterBySelectedEvents);
```